cpp/ql/src/Likely Bugs/Conversion/CastArrayPointerArithmetic.ql

```diff
@@ -35,6 +38,17 @@ class CastToPointerArithFlow extends DataFlow::Configuration {
   }
 }
 
+/**
+ * Holds if the type of `e` is a `DerivedType` with `base` as its base type.
+ *
+ * This predicate ensures that joins go from `e` to `base` instead
+ * of the other way around.
+ */
+pragma[inline]
+predicate hasBaseType(Expr e, Type base) {
+  pragma[only_bind_into](base) = pragma[only_bind_out](e.getType().(DerivedType).getBaseType())
```

Comment:
`pragma[only_bind_into](x) = pragma[only_bind_out](y)` is completely equivalent to both
`x = pragma[only_bind_out](y)` and `pragma[only_bind_into](x) = y`, so you can just choose one. They both express "if you have `y` then you also have `x`". I.e. one of your two pragmas are superfluous.

Reply:
Oh, right. I had blindly copy-pasted the join-order trick from `getNodeEnclosingCallable` in `DataFlowImplCommon` without paying attention to the fact that this was an equality operation. Thanks for spotting this!

Reply:
Fixed in fcd69a005f4e533013843b3ffd283049c2a49f1b.